src/data.js

```diff
@@ -263,45 +263,154 @@ let Artifact = Entity.configure({
+    // deprecated s3 or azure storageType which will never have checking for
+    // presence
+    return {
+      taskId: item.taskId,
+      runId: item.runId,
+      name: item.name,
+      storageType: item.storageType,
+      contentType: item.contentType,
+      details: item.details,
+      expires: item.expires,
+      present: 1,
+    };
   },
   context: [
-    'blobStore',      // BlobStore instance wrapping Azure Blob Storage
-    'privateBucket',  // Private artifact bucket wrapping S3
-    'publicBucket',   // Public artifact bucket wrapping S3
-    'monitor',        // base.monitor instance
+    'blobStore', // BlobStore instance wrapping Azure Blob Storage
+    'privateBucket', // Private artifact bucket wrapping S3
+    'publicBucket', // Public artifact bucket wrapping S3
+    'monitor', // base.monitor instance
+    's3', // Mapping of aws region to aws sdk object
   ],
 });
 
 /** Return JSON representation of artifact meta-data */
 Artifact.prototype.json = function() {
   return {
-    storageType:      this.storageType,
-    name:             this.name,
-    expires:          this.expires.toJSON(),
-    contentType:      this.contentType,
+    storageType: this.storageType,
+    name: this.name,
+    expires: this.expires.toJSON(),
+    contentType: this.contentType,
   };
 };
 
+// Make sure that our hash digests are valid hex
+let validMD5 = /^[a-fA-F0-9]{32}$/;
+let validSHA256 = /^[a-fA-F0-9]{64}$/;
+
+/**
+ * A light wrapper around the create function that ensures some basic things
+ * are correct about the blobs, as well as forcing the present property to be 1
+ * for non-blob storageTypes and converting the truthiness of the present
+ * property to 0 or 1
+ *
+ * NOTE: I have included code here in comments which should be uncommented
+ * when we wish to bar creation of new s3 or azure entities
+ */
+Artifact.create = function(properties, overwriteIfExists /*, allowDeprecated=false*/) {
+  let props = _.default({}, properties);
+  /*
+  if (!allowDeprecated) {
+    if (props.storageType === 's3' || props.storageType === 'azure') {
+      throw new Error('You cannot create new Artifacts of a deprecated storageType');
+    }
+  }
+  */
+
+  // For non-blob storageType entries, we want to force the present value to be
+  // 1 because all other artifacts are either atomically stored in Azure Tables
+  // or are stored in deprecated storageTypes which will never have a check for
+  // present status.  For blob storage items, we'll have as a convenience the
+  // ability to support specifying the present field as either a number or a
+  // boolean.  We will only accept primatives number or boolean
+  if (props.storageType === 'blob' && typeof props.present === 'number') {
+    assert(props.present === 0 || props.present === 1, 'Present must be 0 or 1');
+    props.present = props.present
+  } else if (props.storageType === 'blob' && typeof props.present === 'boolean') {
+    props.present = props.present ? 1 : 0;
```

Comment:
Let's not do this... This won't work in `.modify`...
I agree that it's not pretty to use `1` and `0` as booleans. So if you don't want to do that, just add a type to `azure-entities`...
No reason we can't invent `Entity.types.Bool`. Azure does support booleans `Edm.Boolean`, they are just encoded as booleans.
We would need a test for the boolean type to make sure that filtering works, but other than that it should be pretty straight forward to add a new type to `azure-entities`